03.expression_tree/ExpressionTrees.Task1.ExpressionsTransformator/ChangeParameterVisitor.cs

```diff
@@ -0,0 +1,43 @@
+﻿using System.Collections.Generic;
+using System.Linq.Expressions;
+
+namespace ExpressionTrees.Task1.ExpressionsTransformer
+{
+    public class ChangeParameterVisitor : ExpressionVisitor
+    {
+        private Dictionary<string, object> _parameters;
+
+        public Expression<T> ChangeParameter<T>(Expression<T> expression, Dictionary<string, object> parameters)
+        {
+            if(parameters != null)
+            {
+                _parameters = parameters;
+            }
+
+            return base.VisitAndConvert(expression, "");
+        }
+
+        protected override Expression VisitLambda<T>(Expression<T> node)
+        {
+            var result = base.Visit(node.Body);
+
+            if (result != null)
+            {
+                return Expression.Lambda(result, node.Parameters);
+            }
+
+            return base.VisitLambda(node);
+        }
+
+        protected override Expression VisitParameter(ParameterExpression node)
+        {
+            if (_parameters.TryGetValue(node.Name, out var value) || _parameters != null)
```

Comment:
I'm not sure I understand the logic of this condition. If `_parameters` is null, you will get null ref exception here anyway. If the goal was to check that dictionary is not null, you probably should place the null check in the beginning of the condition, and then use `&& _parameters.TryGetValue...`. However, I think the right place to check it is not in this method, but when you call this visitor.